core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +110,211 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	t.Parallel()
+	for seed := uint64(0); seed < 1_000; seed++ {
+		seed := seed
+		t.Run(fmt.Sprintf("seed_%d", seed), func(t *testing.T) {
+			t.Parallel()
+			testBodyRLPBackwardsCompatibility(t, seed)
+		})
+	}
+}
+
+//nolint:thelper
+func testBodyRLPBackwardsCompatibility(t *testing.T, seed uint64) {
+	rng := ethtest.NewPseudoRand(seed)
+
+	body := &Body{
+		// As this is a round-trip test, the slices MUST NOT be nil as the rlp
+		// package will always set them.
+		Transactions: []*Transaction{},
+		Uncles:       []*Header{},
+	}
+	for i, n := 0, rng.Intn(5); i < n; i++ {
+		tx := NewTx(&LegacyTx{
+			Nonce: rng.Uint64(),
+		})
+		body.Transactions = append(body.Transactions, tx)
+	}
+	for i, n := 0, rng.Intn(5); i < n; i++ {
+		body.Uncles = append(body.Uncles, &Header{
+			ParentHash: rng.Hash(),
+		})
+	}
+
+	var withdrawals Withdrawals
+	for i, n := 0, rng.Intn(4)+1; i < n; i++ {
+		withdrawals = append(withdrawals, &Withdrawal{
+			Index: rng.Uint64(),
+		})
+	}
+
+	tests := []struct {
+		name        string
+		withdrawals []*Withdrawal
+	}{
+		{
+			name: "with_nil_Withdrawals",
+		},
+		{
+			name: "with_empty_Withdrawals",
+			// As the struct field has an `rlp:"optional"` tag, we need to
+			// differentiate between nil (nothing added to RLP) and an empty
+			// slice (empty list added to RLP).
+			withdrawals: []*Withdrawal{},
+		},
+		{
+			name:        "with_Withdrawals",
+			withdrawals: withdrawals,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			body.Withdrawals = tt.withdrawals
+
+			// The original [Body] doesn't implement [rlp.Encoder] nor
+			// [rlp.Decoder] so we can use a methodless equivalent as the gold
+			// standard.
+			type withoutMethods Body
+			wantRLP, err := rlp.EncodeToBytes((*withoutMethods)(body))
+			require.NoErrorf(t, err, "rlp.EncodeToBytes([%T with methods stripped])", body)
+
+			t.Run("Encode", func(t *testing.T) {
+				got, err := rlp.EncodeToBytes(body)
+				require.NoErrorf(t, err, "rlp.EncodeToBytes(%T)", body)
+				assert.Equalf(t, wantRLP, got, "rlp.EncodeToBytes(%T)", body)
+			})
+
+			t.Run("Decode", func(t *testing.T) {
+				got := new(Body)
+				err := rlp.DecodeBytes(wantRLP, got)
+				require.NoErrorf(t, err, "rlp.DecodeBytes(..., %T)", got)
```

Comment:
nit that would be more useful I think?

```suggestion
				require.NoErrorf(t, err, "rlp.DecodeBytes(%v, %T)", wantRLP, got)
```